File: SmartBrowser_java_eclipse/src/com/teuskim/sbrowser/SourceView.java
package com.teuskim.sbrowser;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.ScrollView;
import android.widget.TextView;

public class SourceView extends ScrollView {
	
	private TextView mTextSource;

	public SourceView(Context context, AttributeSet attrs, int defStyle) {
		super(context, attrs, defStyle);
		init(context);
	}

	public SourceView(Context context, AttributeSet attrs) {
		super(context, attrs);
		init(context);
	}

	public SourceView(Context context) {
		super(context);
		init(context);
	}
	
	private void init(Context context){
		LayoutInflater.from(context).inflate(R.layout.source_view, this);
		mTextSource = (TextView) findViewById(R.id.text_source);
	}
	
	public void setSource(String source){
		mTextSource.setText(source);
	}
	
	public String getSource(){
		return mTextSource.getText().toString();
	}
	
}

File: SmartBrowser_kotlin/app/src/main/java/com/teuskim/sbrowser/SourceView.kt
package com.teuskim.sbrowser

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.ScrollView
import android.widget.TextView

class SourceView : ScrollView {

    private var mTextSource: TextView? = null

    var source: String
        get() = mTextSource!!.text.toString()
        set(source) {
            mTextSource!!.text = source
        }

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle) {
        init(context)
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init(context)
    }

    constructor(context: Context) : super(context) {
        init(context)
    }

    private fun init(context: Context) {
        LayoutInflater.from(context).inflate(R.layout.source_view, this)
        mTextSource = findViewById<TextView>(R.id.text_source)
    }

}
